src/txn/txn_rollback_to_stable.c

```diff
@@ -156,6 +158,16 @@ __txn_abort_row_replace_with_hs_value(
 
     WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
 
+    /* Get the full update value from the data store. */
+    unpack = &_unpack;
+    __wt_row_leaf_value_cell(session, page, rip, NULL, unpack);
+    WT_CLEAR(ds_value);
+    if (!__wt_row_leaf_value(page, rip, &ds_value))
+        WT_ERR(__wt_page_cell_data_ref(session, page, unpack, &ds_value));
+
+    WT_CLEAR(full_value);
+    WT_ERR(__wt_buf_set(session, &full_value, ds_value.data, ds_value.size));
```

Comment:
I would have expected this to be something like:
```
    /* Get the full update value from the data store. */
    WT_CLEAR(ds_value);
    if (!__wt_row_leaf_value(page, rip, &ds_value)) {
        unpack = &_unpack;
        __wt_row_leaf_value_cell(session, page, rip, NULL, unpack);
        WT_ERR(__wt_page_cell_data_ref(session, page, unpack, &ds_value));
    }
```
it's a bit less work in the simple case.

Reply:
Changes made.